python/tank/authentication/web_login_support.py

```diff
@@ -0,0 +1,55 @@
+"""
+Web-based login is required when the target Shotgun site uses SSO or Autodesk
+Identity for authentication.
+
+This flag will control if the Unified Login Flow is to be used or not. By
+default, it will not be used.
+
+Unfortunately, not all DCCs/Scripts/Applications that use the Toolkit have all
+the required dependencies to fully support Web-based authentication and renewal.
+This can be a problem for programs not started by the Shotgun Desktop or if
+the Shotgun Desktop is closed after. When the program attemps to authenticate
+or renew a session against a Shotgun site that uses SSO/Autodesk Identity, it
+will fail to do so and will not provide meaningful feedback to the user.
+
+The requirements are:
+- Full support for WebKit/WebEngine with access to the cookie store,
+- SSL support
+- TLS v1.2 support
+- NTLM patch
+
+Some of these could be checked at run-time, but others (TLS v1.2 support,
+NTLM patch) are more difficult to detect.
+
+Unless this flag is explicitely set to True by the enclosing program, we
+will take for granted that the full web login flow is not supported.
+
+WARNING: while SSO login has been supported since tk-core 0.18.151/Shotgun
+Desktop 1.5.3, it was possible for a DCC to be unable to authenticate or renew
+a session. This flag will not change that behaviour.
+
+At this time, only the Shotgun Desktop and Shotgun Create fully support the
+Web based authentication.
+"""
+shotgun_authenticator_support_web_login = False
```

Comment:
I think this _may_ be an issue with core swap.

Reply:
Yes, the flag needs to be carried over to the new core, but since we the flag is not currently used by the code, it's okay I guess?

Reply:
see comment above - the intent with my comment is also a general heads-up. If we need to keep this global state, please add comment to help aid future maintainers of the code!